File: src/main/java/vacanciesalert/model/hhSearchResponse/Vacancy.java
package vacanciesalert.model.hhSearchResponse;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.Instant;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@ToString
public class Vacancy {
    @JsonProperty("id")
    private String id;
    @JsonProperty("name")
    private String name;
    @JsonProperty("area")
    private Area area;
    @JsonProperty("salary")
    private Salary salary;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ")
    @JsonProperty("published_at")
    private Instant publishedAt;
    @JsonProperty("alternate_url")
    private String alternateUrl;
}

File: src/main/kotlin/vacanciesalert/ext/hh/search/Vacancy.kt
package vacanciesalert.ext.hh.search

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

class Vacancy {

    @JsonProperty("id")
    val id: String? = null

    @JsonProperty("name")
    val name: String? = null

    @JsonProperty("area")
    val area: Area? = null

    @JsonProperty("contacts")
    val contacts: Contact? = null

    @JsonProperty("experience")
    val experience: Experience? = null

    @JsonProperty("department")
    val department: Department? = null

    @JsonProperty("employer")
    val employer: Employer? = null

    @JsonProperty("salary")
    val salary: Salary? = null

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ")
    @JsonProperty("published_at")
    val publishedAt: Instant? = null

    @JsonProperty("alternate_url")
    val alternateUrl: String? = null
}
